server/test/integration/vitalsTest/checkForInternalProblemsTest.js

```diff
@@ -0,0 +1,420 @@
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('duration alerts should not be counted, even if duration alerts exceed maxStillnessAlerts in set interval of time for a location', () => {
+    beforeEach(async () => {
+      // Insert a single location and maxStillnessAlerts + 3 sessions
+      const client = await factories.clientDBFactory(db, { isSendingAlerts: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isSendingAlerts: true,
+      })
+      for (let i = 1; i <= maxStillnessAlerts + 1; i += 1) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: ALERT_TYPE.SENSOR_DURATION,
+          incidentCategory: ALERT_TYPE.SENSOR_DURATION,
+        })
+      }
+      for (let i = maxStillnessAlerts + 2; i <= maxStillnessAlerts + 3; i += 1) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: ALERT_TYPE.SENSOR_STILLNESS,
+          incidentCategory: ALERT_TYPE.SENSOR_STILLNESS,
+        })
+      }
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should not send any alerts to Sentry', () => {
+      expect(helpers.logSentry).to.not.have.been.called
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('should log to Sentry after finding more than maxStillnessAlerts have occurred in set interval of time from a single location', () => {
+    beforeEach(async () => {
+      // Insert a single location and maxStillnessAlerts + 1 sessions
+      const client = await factories.clientDBFactory(db, { isSendingAlerts: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isSendingAlerts: true,
+      })
+
+      this.alertType = ALERT_TYPE.SENSOR_STILLNESS
+      this.incidentCategory = ALERT_TYPE.SENSOR_STILLNESS
+      for (let i = 1; i <= maxStillnessAlerts + 1; i += 1) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: this.alertType,
+          incidentCategory: this.incidentCategory,
+        })
+      }
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should send one max stillness alert specific to the location that has exceeded the maxStillnessAlerts in set interval of time to Sentry', () => {
+      expect(helpers.logSentry).to.have.been.calledOnceWithExactly(
+        `Unusually frequent number of stillness alerts (${maxStillnessAlerts + 1}) have been received at ${this.testLocation.locationid}`,
+      )
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('alerts sent earlier than set interval of time prior should not be counted, even if the stillness alerts exceed maxStillnessAlerts', () => {
```

Comment:
The "even if the stillness alerts exceed maxStillnessAlerts" part of this test name makes me think that you this one _should_ send a Sentry alert and log an error. What is it that you wanted to test?

Reply:
i was trying to ensure that stillness alerts sent outside of the interval don't get counted, so that no sentry alert gets logged as a result! 

the scenario i wanted to avoid by making this test was having a stillness alert that was sent earlier than let's say 24 hours still contributing to the number of stillness alerts being counted in this period, which would falsely cause a sentry alert to be logged

i'll reword this to be more clear about it!